File: flutter-external/lib/models/step_result.dart
/// Captures the full request/response lifecycle of a single step.
class StepResult {
  final int step;
  final String name;
  final Map<String, String> requestHeaders;
  final String requestBodyPlaintext;
  final String requestBodyEncrypted;
  final Map<String, String> responseHeaders;
  final String responseBodyEncrypted;
  final String responseBodyDecrypted;
  final int durationMs;
  final bool success;
  final String? error;

  StepResult({
    required this.step,
    required this.name,
    this.requestHeaders = const {},
    this.requestBodyPlaintext = '',
    this.requestBodyEncrypted = '',
    this.responseHeaders = const {},
    this.responseBodyEncrypted = '',
    this.responseBodyDecrypted = '',
    this.durationMs = 0,
    this.success = false,
    this.error,
  });
}

File: dotnet-external/Models/StepResult.cs
namespace IdentityExample.Models;

/// <summary>
/// Result of a single step in the 6-step OAuth2 journey.
/// </summary>
public sealed record StepResult(
    int Step,
    string Name,
    Dictionary<string, string> RequestHeaders,
    string RequestBodyPlaintext,
    string RequestBodyEncrypted,
    Dictionary<string, string> ResponseHeaders,
    string ResponseBodyEncrypted,
    string ResponseBodyDecrypted,
    long DurationMs,
    bool Success,
    string? Error
);
